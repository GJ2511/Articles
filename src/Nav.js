/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { NavLink } from "react-router-dom";


function Nav({loggedInUser}) {
    return (
        <nav className="navbar navbar-expand navbar-dark bg-dark">
            <a className="navbar-brand" href="#">Articles Demo Application</a>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarsExample02" aria-controls="navbarsExample02" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>

            <div className="collapse navbar-collapse" id="navbarsExample02">
                <ul className="navbar-nav ml-auto">
                    {loggedInUser ?
                        <li className="nav-item">
                            <NavLink className="nav-link" to="/signout">Sign Out </NavLink >
                        </li>
                    :
                        <>
                            <li className="nav-item">
                                <NavLink className="nav-link" to="/signin">Sign In </NavLink >
                            </li>
                            <li className="nav-item">
                                <NavLink className="nav-link" to="/signup">Sign Up</NavLink >
                            </li>
                        </>
                    }
                </ul>
            </div>
        </nav>
    )
}

export default Nav;
