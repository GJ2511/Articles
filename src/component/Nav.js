import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

function Nav({ isAuthenticated, signOutRequested }) {
    return (
        <nav className="navbar navbar-expand navbar-dark bg-info">
            <NavLink className="nav-link text-white" to="/" exact strict>
                Articles Demo Application
            </NavLink>
            <button
                className="navbar-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#navbarsExample02"
                aria-controls="navbarsExample02"
                aria-expanded="false"
                aria-label="Toggle navigation"
            >
                <span className="navbar-toggler-icon"></span>
            </button>

            <div className="collapse navbar-collapse" id="navbarsExample02">
                <ul className="navbar-nav ml-auto text-white">
                    {isAuthenticated ? (
                        <>
                            <li className="nav-item">
                                <NavLink className="nav-link" to="/new/article">
                                    Create New Article{' '}
                                </NavLink>
                            </li>
                            <li className="nav-item">
                                <span href="#" className="nav-link" onClick={signOutRequested}>
                                    Sign Out{' '}
                                </span>
                            </li>
                        </>
                    ) : (
                        <>
                            <li className="nav-item">
                                <NavLink className="nav-link" to="/signin">
                                    Sign In{' '}
                                </NavLink>
                            </li>
                            <li className="nav-item">
                                <NavLink className="nav-link" to="/signup">
                                    Sign Up
                                </NavLink>
                            </li>
                        </>
                    )}
                </ul>
            </div>
        </nav>
    );
}

Nav.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    signOutRequested: PropTypes.func.isRequired,
};

export default Nav;
