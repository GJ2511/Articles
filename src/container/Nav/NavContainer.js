import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Nav from '../../component/Nav';
import { signOutRequested } from '../../ducks';

class NavContainer extends Component {
    render() {
        const { authenticated, signOutRequested } = this.props;

        return <Nav isAuthenticated={authenticated} signOutRequested={signOutRequested} />;
    }
}

NavContainer.propTypes = {
    authenticated: PropTypes.bool.isRequired,
    signOutRequested: PropTypes.func.isRequired,
};

const mapStateToProps = ({ applicationReducer }) => ({
    authenticated: applicationReducer.authenticated,
});

export default connect(mapStateToProps, { signOutRequested })(NavContainer);
