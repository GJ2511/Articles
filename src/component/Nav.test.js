import React from 'react';

import Nav from './Nav';

const props = {
    isAuthenticated: true,
    signOutRequested: jest.fn(),
};

describe('<Nav tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<Nav {...props} />)).toMatchSnapshot();
    });
});
