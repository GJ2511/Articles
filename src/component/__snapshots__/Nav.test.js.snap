// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav tests /> match snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-info"
>
  <NavLink
    className="nav-link text-white"
    exact={true}
    strict={true}
    to="/"
  >
    Articles Demo Application
  </NavLink>
  <button
    aria-controls="navbarsExample02"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarsExample02"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarsExample02"
  >
    <ul
      className="navbar-nav ml-auto text-white"
    >
      <li
        className="nav-item"
      >
        <NavLink
          className="nav-link"
          to="/new/article"
        >
          Create New Article
           
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <span
          className="nav-link"
          href="#"
          onClick={[MockFunction]}
        >
          Sign Out
           
        </span>
      </li>
    </ul>
  </div>
</nav>
`;
